### PR TITLE
Doctrine cleanup

### PR DESCRIPTION
**Pull request by zhexu14**
This PR:

- Refactors the doctrine class to have a bit more structure, in anticipation of adding more elements to Doctrine.
- Moves previously hard coded helo-specific altitudes into the Doctrine class, aligning a bunch of altitudes ~200ft in the process.
- Refactors ingress_altitude to combat_altitude to clarify that the altitude is applied to multiple waypoint types, not just the ingress altitude.

Marking as a draft PR as this one should probably be held back until after the 10.0 release.

### CODE REVIEW

**Comment by DanAlbert** [comment]
I created a branch for 10, so we can merge this now. I won't have time to review today unfortunately, but I expect to get to it tomorrow.

File: game/data/doctrine.py [hunk 1]
```diff
@@ -32,72 +32,146 @@ def from_dict(data: dict[str, float]) -> GroundUnitProcurementRatios:
...
-@dataclass(frozen=True)
-class Doctrine:
-    name: str
-
-    cas: bool
-    cap: bool
-    sead: bool
-    strike: bool
-    antiship: bool
+@dataclass
```
Comment (DanAlbert) [inline game/data/doctrine.py:35]: (non-blocking, can deal with it later)

Might as well make this `frozen=True`. Aside from being less bug prone, it allows dataclass to autogenerate the `__eq__` and `__hash__` methods.

File: game/ato/flightplans/waypointbuilder.py [hunk 2]
```diff
@@ -168,10 +165,7 @@ def hold(self, position: Point) -> FlightWaypoint:
             "HOLD",
             FlightWaypointType.LOITER,
             position,
-            # Bug: DCS only accepts MSL altitudes for the orbit task and 500 meters is
```
Comment (DanAlbert) [inline game/ato/flightplans/waypointbuilder.py:171]: I don't think this ever got filed. Could you preserve the comment, or file the bug?

**Review by DanAlbert — APPROVED** [review]

File: game/ato/flightplans/waypointbuilder.py (hunk 2 above)
Comment (zhexu14) [reply game/ato/flightplans/waypointbuilder.py:171]: The bug is filed as #3324 .

File: game/data/doctrine.py (hunk 1 above)
Comment (zhexu14) [reply game/data/doctrine.py:35]: HNY and thanks for reviewing this PR, I'll add this the next time I touch the Doctrine class.

**Pull request merged**